### ISSUE TITLE
General Issue: create_blogpost.R checks for and installs packages 

### ISSUE THREAD

**Issue by bms63** [pharmaverse/blog#245]
### Background Information

@StefanThoma and @manciniedoardo what do you all think about check and installing packages when you create a blog post.  could just be another helper file that create_blogpost calls??

## packages to install

![image](https://github.com/user-attachments/assets/ba18dd06-7403-46bc-a7c9-8e8c69052036)

## Helper program to check and install packages

check_and_install <- function(packages) {
  for (package in packages) {
    if (!require(package, character.only = TRUE)) {
      install.packages(package, dependencies = TRUE)
      library(package, character.only = TRUE)
    } else {
      message(paste(package, "already installed"))
    }
  }
  message("Let's program some ADaMs!")
}

# Use the function
check_and_install(c(
  packages here...
))

**Comment by manciniedoardo** [comment]
sounds like a great idea @bms63 !

**Comment by bms63** [comment]
god...teal has so many dependencies...

**Comment by StefanThoma** [comment]
Why not `renv`?

**Comment by bms63** [comment]
I am always weary of `renv` it made working on admiral unnecessarily challenging for new folks.

having just installed teal and mirai and I wondering if there is a way to not have to install them for new posers - just takes forever.  guess the docker image could be handy????

**Comment by StefanThoma** [comment]
I wonder if there is a way where only a select few blogposts render, plus the new one?
That way they can test their blog, and when uploaded everything renders.

**Comment by manciniedoardo** [comment]
Yeah I always temporarily delete ~40 posts to render the website!

**Comment by pawelru** [comment]
I see two options: [`r-lib/actions/setup-r-dependencies`](https://github.com/r-lib/actions/tree/v2-branch/setup-r-dependencies) OR [`r-lib/actions/setup-renv`](https://github.com/r-lib/actions/tree/v2-branch/setup-renv)

`r-lib/actions/setup-r-dependencies`:
- this requires a DESCRIPTION file to read the dependencies from. You might create a dummy DESC file - it does not have to become a _package_. We are currently doing this trick in [`pharmaverse/examples`](https://github.com/pharmaverse/examples). Please have a look at its [`DESCRIPTION` file](https://github.com/pharmaverse/examples/blob/main/DESCRIPTION) and how the [action](https://github.com/pharmaverse/examples/blob/main/.github/workflows/docs.yaml) look like and its [results](https://github.com/pharmaverse/examples/actions/runs/11610230165/job/32329102584).
- this is super fast because it is using PPM and install from binaries (instead of from source) plus it has in-build caching. Once downloaded, it would stay in cache and you won't repeat the download (unless there is a new version or cache removed). Have a look at the latest result [here](https://github.com/pharmaverse/examples/actions/runs/11610230165/job/32329102584#step:6:8321) - `ℹ No downloads are needed`.

`r-lib/actions/setup-renv`:
This unfortunately I have never used so can't share my experience. I just know that it exists.

By any means - I do not recommend to maintain a list of packages with its dependencies (!). Some of those packages are still under development and the dependencies are in scope of a change. Let it be identified automatically. There are tools for that. And let it be a low maintenance thing from our side. (That's for `mirai` package discussed here - you should not care about that :) )
The list you showed is of good size to be putted in the dummy DESC file. I have all the positive experience with `setup-r-dependencies` so I would suggest this.

**Comment by bms63** [comment]
Thanks @pawelru - I agree we should move to a package-like structure and take advantage of some of these r-libs actions.

I am also interested in making blog contributors have a really easy time writing a post and looking at ways to help minimize setup and headaches for them.  I am assuming a completely newb-like person here!
